src/contracts/atlas/GasAccounting.sol

```diff
@@ -237,71 +249,167 @@ abstract contract GasAccounting is SafetyLocks {
+        uint256 gasUsed = (gasWaterMark + _SOLVER_BASE_GAS_USED - gasleft()) * tx.gasprice;
 
         if (includeCalldata) {
             gasUsed += _getCalldataCost(solverOp.data.length);
         }
 
-        gasUsed = (gasUsed + ((gasUsed * SURCHARGE_RATE) / SURCHARGE_SCALE));
-        _assign(solverOp.from, gasUsed, false, bidFind);
+        gasUsed = (gasUsed * ((SURCHARGE_SCALE + ATLAS_SURCHARGE_RATE + BUNDLER_SURCHARGE_RATE) / SURCHARGE_SCALE));
+
+        // Calculate what the solver owes
+        // NOTE: This will cause an error if you are simulating with a gasPrice of 0
+        if (!bidFind && !result.updateEscrow()) {
+            // CASE: Solver is not responsible for the failure of their operation, so we blame the bundler
+            // and reduce the total amount refunded to the bundler
+            writeoffs += gasUsed;
+        } else {
+            // CASE: Solver failed, so we calculate what they owe.
+            uint256 deficit = _assign(solverOp.from, gasUsed, false, bidFind);
+            if (deficit > 0) {
+                // Write off any deficit as a gas loss to the bundler so that other solvers aren't forced to pay it.
+                writeoffs += deficit;
+            }
+        }
+    }
+
+    /// @param ctx Context struct containing relavent context information for the Atlas auction.
+    /// @param solverGasLimit The maximum gas limit for a solver, as set in the DAppConfig
+    function _getAdjustedClaimsAndWriteoffs(
+        Context memory ctx,
+        uint256 solverGasLimit
+    )
+        internal
+        view
+        returns (uint256 _claims, uint256 _writeoffs)
+    {
+        _claims = claims;
+        _writeoffs = writeoffs;
+
+        uint256 _gasLeft = gasleft(); // Hold this constant for the calculations
+
+        uint256 _gasRemainder =
+            _gasLeft * tx.gasprice * (SURCHARGE_SCALE + ATLAS_SURCHARGE_RATE + BUNDLER_SURCHARGE_RATE) / SURCHARGE_SCALE;
+        _claims -= _gasRemainder;
+
+        // By reducing the _claims, Solvers end up paying less in total.
+        if (ctx.solverCount > 0) {
+            uint256 _bundlerGasOveragePenalty;
+
+            // Calculate the unadjusted bundler gas surcharge
+            uint256 _grossBundlerGasSurcharge =
+                (_claims * BUNDLER_SURCHARGE_RATE) / (SURCHARGE_SCALE + ATLAS_SURCHARGE_RATE + BUNDLER_SURCHARGE_RATE);
+
+            // Calculate an estimate for how much gas should be remaining
+            // NOTE: There is a free buffer of one SolverOperation because solverIndex starts at 0.
+            uint256 _upperGasRemainingEstimate =
+                (solverGasLimit * (ctx.solverCount - ctx.solverIndex)) + _BUNDLER_GAS_PENALTY_BUFFER;
+
+            // Increase the _writeoffs value if the bundler set too high of a gas parameter and forced solvers to
+            // maintain higher escrow balances.
+            if (_gasLeft > _upperGasRemainingEstimate) {
+                // Penalize the bundler's gas
```

Comment:
*reduce the bundlers gas surcharge payment proportionally relative to how much they over-estimated the gasLimit

Reply:
While it's functionally very similar, we explicitly want to make sure we're increasing the bundler's refund's offset / contra account (_writeoffs) rather than decreasing the payable amount.  This is because of the way the surcharges are already built into `_claims` and the other accounting variables.